src/components/chooseFolder/chooseFolder.js

```diff
@@ -1,24 +1,66 @@
 import React, { Component } from "react";
+import './../checkBoxSelection/checkBoxSelection.scss';
+
+import CheckBoxSelection from "../checkBoxSelection/checkBoxSelection";
 class chooseFolder extends Component {
-  constructor(props) {
+ constructor(props) {
     super(props);
     this.state = {
-      path: []
-    };
+      path:[],
+      arr:null,
+       displayMenu: false,
+    }
+  } 
+
+  showDropdownMenu = event => {
+    event.preventDefault();
+    this.setState({ displayMenu: true }, () => {
+    document.addEventListener('click', this.hideDropdownMenu);
+    });
   }
 
+  hideDropdownMenu = () => {
+    this.setState({ displayMenu: true }, () => {
+      document.removeEventListener('click', this.hideDropdownMenu);
+    });
+
+  }
   onChangeHandler = event => {
     this.setState({ path: event.target.files });
   };
 
   render() {
-    for (var item of this.state.path) {
-      console.log(item);
-    }
+   const arr = [];
+   for (var key in this.state.path) {
+      if(key<this.state.path.length){
```

Comment:
since you are using a for each statement you dont need to check that the key is still in the range of the array, you wont ever get a out of bounds exception